File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.java
class Solution {
    public int maxProfit(int[] prices) {
        int maxProfit = 0;
        int minVal = Integer.MAX_VALUE;
        int n = prices.length;
        for(int i=0;i<n;i++){
            minVal = Math.min(minVal,prices[i]);
            maxProfit = Math.max(maxProfit,prices[i]-minVal);
        }
        return maxProfit;
    }
}

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int n = static_cast<int>(prices.size());
        int min = prices[0];
        int current_max = prices[0];
        int ans = 0;
        for(int i=1;i<n;i++){
            if(prices[i]<current_max){
                ans += current_max - min;
                min = prices[i];
            }
            current_max = prices[i];
        }
        ans += current_max - min;
        return ans;
    }
};

